### PR TITLE
feat: Simplified Dueling Dags - Mutation Recover - Add mutationID and  lastServerAckdMutationID to Client.

### PR DESCRIPTION
**Pull request by grgbkr**
Add `mutationID` and  `lastServerAckdMutationID` to `Client` and update `initClient` and `persist` to
write them appropriately.  These new fields will be used by other clients to determine if a client has 
pending mutations (persisted local mutations unacknowledged by the server) that it can push on the 
other client's behalf.  We will refer to this process as “mutation recovery”, as one client is recovering the
 mutations of another client, by reading them from the other client’s perdag stage and pushing on the 
other client’s behalf. 

See [Mutation Recovery design](https://www.notion.so/Mutation-Recovery-Avoiding-Mutation-Loss-using-PerDag-state-f54025b52cbc435692abca3307947d15). 

Part of #671

### CODE REVIEW

File: src/mod.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export {Replicache} from './replicache';
 export {TransactionClosedError} from './transaction-closed-error';
 
+export type {Hash} from './hash';
```
Comment (arv) [inline src/mod.ts]: No.

We should not expose this in the public API

File: src/persist/persist.ts [hunk 2]
```diff
@@ -41,21 +42,42 @@ export async function persist(
...
 }
 
 async function gatherTempChunks(
   memdag: dag.Store,
-): Promise<[ReadonlyMap<Hash, dag.Chunk>, Hash]> {
+): Promise<
+  [
+    ReadonlyMap<Hash, dag.Chunk>,
+    Hash,
+    {mutationID: number; lastMutationID: number},
```
Comment (arv) [inline src/persist/persist.ts]: Either make this:

```ts
type _ = [
    map: ReadonlyMap<Hash, dag.Chunk>,
    hash: Hash,
    mutationID: number,
    lastMutationID: number,
]
```

or

```ts
type _ = {
    map: ReadonlyMap<Hash, dag.Chunk>,
    hash: Hash,
    mutationID: number,
    lastMutationID: number,
}
```

mixing object and array seems a bit ugly

Comment (arv) [reply src/persist/persist.ts]: the benefit of tuple over object is that it compresses better. the benefit of object is that it reads better...

File: src/persist/persist.ts [hunk 3]
```diff
@@ -96,13 +118,16 @@ async function writeFixedChunks(
   fixedChunks: FixedChunks,
   mainHeadHash: Hash,
   clientID: string,
+  mutationIDs: {mutationID: number; lastMutationID: number},
```
Comment (arv) [inline src/persist/persist.ts]: same here... just make these into two parameters?

**Review by arv — APPROVED** [review]
But please undo the changes to mod.ts

We should probably add a comment in there or rename the file. The name was picked because mod.ts has become the defacto standard name for the public API in deno.

File: src/mod.ts (hunk 1 above)
Comment (grgbkr) [reply src/mod.ts]: I see.  So just src/mod.ts is the public API, correct?  The other mod.ts files are for cross directory export/import?

File: src/persist/persist.ts (hunk 2 above)
Comment (grgbkr) [reply src/persist/persist.ts]: done

File: src/persist/persist.ts (hunk 3 above)
Comment (grgbkr) [reply src/persist/persist.ts]: done

File: src/mod.ts (hunk 1 above)
Comment (arv) [reply src/mod.ts]: That is correct

**Review by arv — APPROVED** [review]
LGTM

**Review by arv — APPROVED** [review]

**Pull request merged**